x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,574 @@
+passwords for keys or keystores that are stored in the
+<<secure-settings,secure settings>>, then you must complete a
+<<use-rolling-restarts,rolling restart>>. {es} will not reload changes for
+passwords stored in the secure settings. 
+
+[[use-rolling-restarts]]
+.Rolling restarts are preferred
+****
+While it's possible to do an in-place update for security certificates, using
+a <<restart-cluster-rolling,rolling restart>> on your cluster is safer. An in-place update avoids some
+complications of a rolling restart, but incurs the following risks:
+
+* If you use PEM files, your certificate and key are in separate files. You
+must update both files _simultaneously_ or the node might experience a temporary
+period where it cannot establish new connections.
+* Updating the certificate and key does not automatically force existing
+connections to refresh. This means that even if you make a mistake, a node can
+seem like it's functioning but only because it still has existing connections.
+It's possible that a node will be unable to connect with other nodes, rendering
+it unable to recover from a network outage or node restart.
+****
+
+[[update-node-certs-same]]
+=== Update security certificates with the same CA
+++++
+<titleabbrev>With the same CA</titleabbrev>
+++++
+
+If your CA hasn't changed, you can replace your existing certificate on each
+node with a new certificate that has the same filename. When {es} restarts, it
+reloads the file and uses the new certificate. Using the same filename means
+that you don't have to update the `elasticsearch.yml` file, and can just
+replace the existing certificate.
+
+You don't have to restart each node, but doing so forces new TLS connections and is <<use-rolling-restarts,a recommended practice>> when updating certificates.
+Therefore, the following steps include a node restart after updating each
+certificate.
+
+You'll generate new node certificates and keys for both the transport layer and
+the HTTP layer.
+
+[[cert-password-updates]] 
+IMPORTANT: If your keystore is password protected, the password
+is stored in the {es} keystore, _and_ the password needs to change, then you
+must perform a <<restart-cluster-rolling,rolling restart>> on your cluster. You
+must also use a different file name for the keystore so that {es} reloads the
+new file.
+
+This procedure assumes that the you have access to the CA certificate and key
+that was originally generated (or otherwise held by your organization) and used
+to sign the certificate that is currently in use. It also assumes that the
+clients connecting to {es} on the HTTP layer are configured to trust the CA
+certificate.
+
+TIP: If your CA has changed, complete the steps in 
+<<update-node-certs-different,update security certificates with a different CA>>.
+
+[[node-certs-same-transport]]
+==== Generate a new certificate for the transport layer
+Create a new PKCS#12 file that contains a certificate and key and replace
```

Comment:
Again, I find the summary in the beginning that contains a subset/summary of the steps slightly confusing. Can we be very generic and move straight to the concrete steps ?